pandas/core/reshape/tile.py

```diff
@@ -435,11 +437,14 @@ def _bins_to_cuts(
                     "Bin labels must be one fewer than the number of bin edges"
                 )
         if not is_categorical_dtype(labels):
-            labels = Categorical(
-                labels,
-                categories=labels if len(set(labels)) == len(labels) else None,
-                ordered=ordered,
-            )
+            if labels_are_unique:
```

Comment:
why this extra logic?

Reply:
Sorry... I thought I reverted this. Forgot to push.

But, the `set(labels)` stuff is somewhat expensive (don't have the exact numbers right now). There may be cases where we know that we have unique labels and can avoid the check.